### PR TITLE
Update embedfs implementation

### PR DESCRIPTION
**Pull request by this-kirke**
Fixes to embedfs implementation:
- Fix path normalization to support billy's absolute path convention ("/")
- Implement missing Lstat method (was returning ErrNotSupported)
- Add proper embed.FS to billy path conversion throughout all methods

Additional test coverage:
- Add dedicated test files for different functionality areas
- Test File interface methods including ReadAt, Close, Lock/Unlock
- Add path normalization and edge case testing
- Test empty file handling and boundary conditions
- Verify proper error handling for read-only operations

Test infrastructure:
- Create embedfs_testdata package for reusable test data
- Resolve import cycles with clean provider pattern
- Add test data including nested directories and various file types
- Organize tests by functionality (fs, dir, file operations)

The embedfs implementation now provides billy.Filesystem compliance for read-only embedded filesystems.

### CODE REVIEW

File: embedfs_testdata/provider.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+)
+
+//go:embed testdata
+var TestData embed.FS
+
+// GetTestData returns the raw embed.FS for tests to wrap with their own embedfs.New().
+// This avoids import cycles while providing embedded test data.
+func GetTestData() *embed.FS {
+	return &TestData
+}
```
Comment (pjbgf) [inline embedfs_testdata/provider.go:16]:
```suggestion
}

```

**Comment by this-kirke** [comment]
Hi Paulo! Thank you for the review. I've updated with requested changes; all tests pass.

File: embedfs/embed_test.go [hunk 2]
```diff
@@ -8,17 +8,11 @@ import (
...
 	"github.com/stretchr/testify/require"
 )
 
-//go:embed testdata/empty.txt
-var singleFile embed.FS
-
-//go:embed testdata
-var testdataDir embed.FS
-
-var empty embed.FS
```
Comment (pjbgf) [inline embedfs/embed_test.go:21]: We need this added again for `TestRead` or as a func in `provider.go`. Same for `singleFile`.

File: embedfs/embed_test.go [hunk 3]
```diff
@@ -184,30 +178,23 @@ func TestReadDir(t *testing.T) {
...
 		{
-			name:    "empty",
+			name:    "empty path",
 			path:    "",
-			fs:      &empty,
-			want:    []string{},
-			wantErr: true,
+			fs:      testdata.GetTestData(),
+			want:    []string{"testdata"},
+			wantErr: false,
```
Comment (pjbgf) [inline embedfs/embed_test.go:191]: This is changing the meaning of the test. Let's revert this and change `wantErr` to `false`.

```suggestion
			name:    "empty FS",
			path:    "",
			fs:      &empty,
			want:    []string{},
			wantErr: false,
```

File: embedfs/embed_test.go [hunk 4]
```diff
@@ -184,30 +178,23 @@ func TestReadDir(t *testing.T) {
 		wantErr bool
 	}{
 		{
-			name: "singleFile",
+			name: "testdata",
 			path: "testdata",
-			fs:   &singleFile,
-			want: []string{"empty.txt"},
+			fs:   testdata.GetTestData(),
```
Comment (pjbgf) [inline embedfs/embed_test.go:183]: This needs to be changed to an embedFS which returns a single file.

File: embedfs/embed_test.go [hunk 5]
```diff
@@ -184,30 +178,23 @@ func TestReadDir(t *testing.T) {
...
-			fs:   &testdataDir,
-			want: []string{"empty.txt", "empty2.txt"},
-		},
-		{
-			name:    "testdataDir return no dir names",
-			path:    "",
-			fs:      &testdataDir,
-			want:    []string{},
-			wantErr: true,
+			name: "root path",
```
Comment (pjbgf) [inline embedfs/embed_test.go:194]: This is a different sue case then the previous two, let's add them back up and combine with this new addition.

File: embedfs/embed_test.go [hunk 6]
```diff
@@ -338,10 +325,188 @@ func TestJoin(t *testing.T) {
...
+	f, err := fs.Open("/testdata/file1.txt")
+	require.NoError(t, err)
+	defer f.Close()
+	
+	// Read the actual content
+	buf := make([]byte, 100)
+	n, err := f.Read(buf)
+	require.NoError(t, err)
+	assert.Equal(t, "Hello from embedfs!", string(buf[:n]))
+}
```
Comment (pjbgf) [inline embedfs/embed_test.go:364]: This is redundant - `TestOpen` already opens and reads the contents of the files.

```suggestion
```

File: embedfs/embed_test.go [hunk 7]
```diff
@@ -338,10 +325,188 @@ func TestJoin(t *testing.T) {
...
+	// Test nested file read
+	f, err := fs.Open("/testdata/subdir/nested.txt")
+	require.NoError(t, err)
+	defer f.Close()
+	
+	buf := make([]byte, 100)
+	n, err := f.Read(buf)
+	require.NoError(t, err)
+	assert.Equal(t, "Nested file content", string(buf[:n]))
+}
```
Comment (pjbgf) [inline embedfs/embed_test.go:380]: Please add this as a new testcase for `TestOpen` instead.

```suggestion
```

File: embedfs/embed_test.go [hunk 8]
```diff
@@ -338,10 +325,188 @@ func TestJoin(t *testing.T) {
...
+			// All these should work with our path normalization
+			_, err := fs.Stat(tt.path)
+			if tt.name == "deep file" {
+				require.NoError(t, err, "file should exist")
+			} else {
+				require.NoError(t, err, "directory should exist")
+			}
+		})
+	}
+}
```
Comment (pjbgf) [inline embedfs/embed_test.go:409]: Replace this with a new test cases within `TestStat`, as the coverage is largely the same.

File: embedfs/embed_test.go [hunk 9]
```diff
@@ -338,10 +325,188 @@ func TestJoin(t *testing.T) {
...
+			buf := make([]byte, tt.length)
+			n, err := f.ReadAt(buf, tt.offset)
+			
+			if tt.offset >= 19 {  // Beyond file size
+				require.Error(t, err)
+				assert.Equal(t, 0, n)
+			} else {
+				if tt.offset+int64(tt.length) > 19 {
+					// Partial read at end of file
+					require.Error(t, err) // Should be EOF
```
Comment (pjbgf) [inline embedfs/embed_test.go:445]:
```suggestion
					require.ErrorIs(t, err, io.EOF)
```

File: embedfs/embed_test.go [hunk 10]
```diff
@@ -338,10 +325,188 @@ func TestJoin(t *testing.T) {
...
+	// Test multiple closes (should be safe)
+	err = f.Close()
+	require.NoError(t, err)
+
+	err = f.Close()
+	require.NoError(t, err)
+
+	// Note: embedfs doesn't necessarily fail operations after close
+	// since embed.FS files remain readable. This tests that Close() works
+	// without error, but doesn't enforce post-close failure behavior.
```
Comment (pjbgf) [inline embedfs/embed_test.go:484]:
```suggestion
```

File: embedfs/embed_test.go [hunk 11]
```diff
@@ -338,10 +325,188 @@ func TestJoin(t *testing.T) {
...
+	fs := New(testdata.GetTestData())
+
+	f, err := fs.Open("/testdata/file1.txt")
+	require.NoError(t, err)
+
+	// Test first close
+	err = f.Close()
+	require.NoError(t, err)
+
+	// Test multiple closes (should be safe)
```
Comment (pjbgf) [inline embedfs/embed_test.go:475]:
```suggestion
	// Multiple closes must not cause it to error.
```

File: embedfs/embed_test.go [hunk 12]
```diff
@@ -338,10 +325,188 @@ func TestJoin(t *testing.T) {
...
+	defer f.Close()
+
+	// Lock/Unlock should be no-ops that don't error
+	err = f.Lock()
+	require.NoError(t, err)
+
+	err = f.Unlock()
+	require.NoError(t, err)
+
+	// Multiple lock/unlock sequences should work
```
Comment (pjbgf) [inline embedfs/embed_test.go:503]:
```suggestion
	// Invalid lock/unlock sequences are not checked.
```

File: embedfs/embed_test.go [hunk 13]
```diff
@@ -338,10 +325,188 @@ func TestJoin(t *testing.T) {
...
+	require.NoError(t, err)
+	defer f.Close()
+	
+	buf := make([]byte, 100)
+	n, err := f.Read(buf)
+	require.NoError(t, err)
+	assert.Equal(t, "Nested file content", string(buf[:n]))
+}
+
+func TestEmbedfs_PathNormalization(t *testing.T) {
```
Comment (pjbgf) [inline embedfs/embed_test.go:382]:
```suggestion
func TestPathNormalization(t *testing.T) {
```

File: embedfs/embed_test.go [hunk 14]
```diff
@@ -338,10 +325,188 @@ func TestJoin(t *testing.T) {
...
+	
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// All these should work with our path normalization
+			_, err := fs.Stat(tt.path)
+			if tt.name == "deep file" {
+				require.NoError(t, err, "file should exist")
+			} else {
+				require.NoError(t, err, "directory should exist")
+			}
```
Comment (pjbgf) [inline embedfs/embed_test.go:406]: Can we repurpose this to target the `normalizePath` func instead?

File: embedfs/dir_test.go [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+
+	fs := New(testdata.GetTestData())
+
+	// Test reading nested directories
+	entries, err := fs.ReadDir("/testdata/subdir")
+	require.NoError(t, err)
+	assert.Len(t, entries, 1)
+	assert.Equal(t, "nested.txt", entries[0].Name())
+	assert.False(t, entries[0].IsDir())
+}
```
Comment (pjbgf) [inline embedfs/dir_test.go:25]: This is redundant, maybe we add it as an additional test case in `TestReadDir`.

File: embedfs/dir_test.go [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+	entries, err := fs.ReadDir("/testdata")
+	require.NoError(t, err)
+	
+	// Verify all entries have proper FileInfo
+	for _, entry := range entries {
+		assert.NotEmpty(t, entry.Name())
+		assert.NotNil(t, entry.ModTime())
+		assert.Greater(t, entry.Size(), int64(-1)) // Size can be 0 but not negative
+	}
+}
```
Comment (pjbgf) [inline embedfs/dir_test.go:41]: Instead of having its own test, this could be an improvement for `TestReadDir` - adding expectations around `FileInfo` and verifying them.

File: embedfs/dir_test.go [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+		if entry.Name() == "subdir" {
+			subdirEntry = entry
+			break
+		}
+	}
+	
+	require.NotNil(t, subdirEntry, "subdir should be found")
+	assert.True(t, subdirEntry.IsDir(), "subdir should be a directory")
+	assert.Equal(t, "subdir", subdirEntry.Name())
+}
```
Comment (pjbgf) [inline embedfs/dir_test.go:63]: Move this one to `embed_test.go` and delete this file.

File: embedfs/fs_test.go [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+	n, err := f.Read(buf)
+	require.Error(t, err) // Should be EOF
+	assert.Equal(t, 0, n)
+
+	// Test ReadAt on empty file
+	n, err = f.ReadAt(buf, 0)
+	require.Error(t, err) // Should be EOF
+	assert.Equal(t, 0, n)
+}
+
```
Comment (pjbgf) [inline embedfs/fs_test.go:109]: We can delete this file, these tests seem to be already covered elsewhere.

```suggestion
```

**Review by pjbgf — CHANGES_REQUESTED** [review]
@this-kirke Thanks for the follow up. I just did a full review now, there are a few tests that are redundant which we can remove. But otherwise it LGTM.

Please squash your commits once you are done with the changes below.

**Comment by this-kirke** [comment]
Thank you for the detailed review! I agree on all points. I think everything should be resolved now; please let me know if there's anything else you notice. We can squash on complete.
